frontend/src/app/features/work-packages/routing/wp-view-base/view-services/wp-view-additional-elements.service.ts

```diff
@@ -143,4 +149,37 @@ export class WorkPackageViewAdditionalElementsService {
 
     return Promise.resolve([]);
   }
+
+  private requireWorkPackageShares(wpIds:string[]):Promise<string[]> {
+    if (!this.wpTableColumns.hasShareColumn()) { return Promise.resolve([]); }
+
+    const filters = new ApiV3FilterBuilder()
+      .add('entityType', '=', ['WorkPackage'])
+      .add('entityId', '=', wpIds);
+
+    const workPackageShareRequest = this
+      .apiV3Service
+      .shares
+      .filtered(filters, { pageSize: '-1' })
```

Comment:
Even with pageSize -1, you will never exceed the `Setting.apiv3_max_page_size`. This means by default you'll never be able to get more than 1000 results here. This might be fine, and we do that in a lot of places without a care in the world until we get complaints that some entries are missing.

The frontend does have some mitigations for that, but they need to be actively called e.g., by using the `ApiV3GettableResourceCollection#getPaginatedResults` to actively load all pages.

Reply:
Would I need to do anything special there?